aiida_quantumespresso/workflows/pw/base.py

```diff
@@ -272,26 +272,6 @@ def validate_pseudos(self):
             self.report(f'{exception}')
             return self.exit_codes.ERROR_INVALID_INPUT_PSEUDO_POTENTIALS
 
-    def validate_resources(self):
-        """Validate the inputs related to the resources.
-
-        One can omit the normally required `options.resources` input for the `PwCalculation`, as long as the input
-        `automatic_parallelization` is specified. If this is not the case, the `metadata.options` should at least
-        contain the options `resources` and `max_wallclock_seconds`, where `resources` should define the `num_machines`.
-        """
-        if 'automatic_parallelization' not in self.inputs and 'options' not in self.ctx.inputs.metadata:
-            return self.exit_codes.ERROR_INVALID_INPUT_RESOURCES
-
-        # If automatic parallelization is not enabled, we better make sure that the options satisfy minimum requirements
-        if 'automatic_parallelization' not in self.inputs:
```

Comment:
Looking at the `automatic_parallelization` some more, won't the fact that `num_machines` can't be defined for some schedulers cause problems here? E.g. this line:

https://github.com/aiidateam/aiida-quantumespresso/blob/33cb11e3c9a1f382fac69c15d74a2b4f6d331d4c/aiida_quantumespresso/workflows/pw/base.py#L347

In fact, after reading https://github.com/aiidateam/aiida-quantumespresso/commit/70d97b1cf40e3a6a4d9fe8bf0fccba09a3d4d102, does this feature still work for recent  QE versions? Maybe it would be a good idea to fix this in this PR.

... Just kidding on that last part. 😉

Reply:
> .. Just kidding on that last part. wink

Haha I was going to say, you are giving me a "cookie of my own dough" ;)

There is already an open issue for the automatic parallelization because I doubt it has been working at all for a long time.  There is another problem that I realized. Originally, since the `metadata.options.resources` were not validated on the base `CalcJob` you could submit with them missing and so the `PwBaseWorkChain` could set them instead if automatic parallelization was selected. This is why this validate resource function was even there. Now though, the `PwCalculation` requires it so even if the user _were_ to want to use automatic parallelization, they would still have to define acceptable resources, even though they may be overwritten later. Not the worst problem, since there is a workaround, but also not the nicest. We could replace the validator of the `CalcJob` class but that would be tricky because the resource validation is done in the top level namespace validator and we only want to get rid of part that, meaning we would have to copy paste all the rest of the validator, which would be a nightmare to maintain.

Anyway, I am not to worried about the AP feature here because it was always experimental and don't think it is critical as long as the normal functionality for the base works.

Reply:
Alright, we'll have a closer look at the automatic parallelisation some other time then. 😀